### PR TITLE
Error handling (ZRUTECH-494)

### PR DESCRIPTION
**Pull request by zarnowsk**
@TaylorChristie can we test this somehow?

### CODE REVIEW

**Comment by TaylorChristie** [comment]
The easiest way to test is point the action to dev, and then run it against a broken solution file that will error.

**Review by TaylorChristie — CHANGES_REQUESTED** [review]
- check for failed state and fail the action if it is failed
- check for timeout and fail the action

File: index.js [hunk 1]
```diff
@@ -36,21 +39,19 @@ const deploy = async () => {
...
-						`${base_url}/details/?instanceId=${instanceId}`,
+						`${base_url}/details?instanceId=${instanceId}`,
 						{ headers }
 					);
 					const state = deploymentDetails.data.metadata.status;
 					console.log(`Current state: ${state}`);
 
-					if (i > 100) {
-						console.error("Timing out after ~500 seconds.");
+					if (i > timeout * 10) {
```
Comment (TaylorChristie) [inline index.js]: `i * 10` is the amount of seconds elapsed (roughly). However `timeout * 10` is not the amount of seconds in `timeout` minutes - I believe it is more. 

I think the right calculation is `i * 10 > timeout * 60` or more simply `i > timeout * 6`

**Review by TaylorChristie — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (zarnowsk) [reply index.js]: @TaylorChristie yeah you're right, bust have been late...

File: index.js [hunk 2]
```diff
@@ -62,17 +63,22 @@ const deploy = async () => {
 
 						keepGoing = false;
 						return;
+					} else if (state.toUpperCase() === "FAILED") {
+						throw new Error("Instance failed to deploy in ZRU");
 					}
 				} catch (e) {
-					console.error(e);
+					throw new Error(`Instance deployment failed: ${e}`);
```
Comment (TaylorChristie) [inline index.js:76]: This is actually catching an error from the /details endpoint, which at some points could fail... Lets maybe keep the console.error(e) for now and add a message saying `error fetching details on deployment, retrying`.

Comment (zarnowsk) [reply index.js:76]: Added `trycatch` around the endpoint because line 70 `catch` catches a bunch of different errors

**Review by TaylorChristie — APPROVED** [review]

**Pull request merged**